tracing.go

```diff
@@ -84,53 +87,155 @@ func (s Span) ParentID() uint64 { return s.parentID }
 // SpanID returns the id of this specific RPC
 func (s Span) SpanID() uint64 { return s.spanID }
 
-// EnableTracing controls whether tracing is enabled for this context
-func (s *Span) EnableTracing(enabled bool) {
-	if enabled {
-		s.flags |= tracingFlagEnabled
-	} else {
-		s.flags &= ^tracingFlagEnabled
+// Flags returns flags bitmap. Interpretation of the bits is up to the tracing system.
+func (s Span) Flags() byte { return s.flags }
+
+// SetTraceID sets traceID
+func (s *Span) SetTraceID(traceID uint64) { s.traceID = traceID }
```

Comment:
hm, possibly. The Jaeger Tracer internally has the same methods on some interface to inject span IDs into a carrier played by the `Span` struct. But there's no cross-dependency on the interface name, so as long as we cast to this `injectableSpan` before passing `Span` as a carrier, it should work. Duck typing is funny.